adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralNativeRenderer.m

```diff
@@ -0,0 +1,256 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "GADMAdapterMintegralNativeRenderer.h"
+#import "GADMAdapterMintegralUtils.h"
+#import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDK/MTGBidNativeAdManager.h>
+#include <stdatomic.h>
+
+@interface GADMAdapterMintegralNativeRenderer ()
+<MTGBidNativeAdManagerDelegate,
+GADMediationNativeAd,
+MTGMediaViewDelegate>
+
+@end
+
+@implementation GADMAdapterMintegralNativeRenderer {
+    /// The completion handler to call when the ad loading succeeds or fails.
+    GADMediationNativeLoadCompletionHandler _adLoadCompletionHandler;
+    
+    /// Holds the state for impression being logged.
+    atomic_flag _impressionLogged;
```

Comment:
Can you explain why we need to hold the state for impression?

Reply:
I will delete the code